### ISSUE TITLE
`plot()` and `plot_3d()` of stl imported structure

### ISSUE THREAD

**Issue by tomflexcompute** [flexcompute/tidy3d#1522]
For structures imported from stl files, current `plot()` and `plot_3d()` can either be wrong or not show the structures at all, which led to some user frustration. Could this be fixed in the near future?

A side note that in the web GUI, the structure actually shows up mostly correctly.

**Comment by momchil-flex** [comment]
Who worked on this before? @caseyflex @lucas-flexcompute @tylerflex ?

@tomflexcompute it would be good to have an example of the plotting being wrong.

**Comment by tomflexcompute** [comment]
I think we can use this notebook to test https://www.flexcompute.com/tidy3d/examples/notebooks/90OpticalHybrid/

**Comment by lucas-flexcompute** [comment]
I can't see any errors in `plot()` in either develop or pre/2.7 branches. Is there a particular case that is not working?

The geometry if completely missing in `plot_3d()`, indeed, but that's [generated externally](https://github.com/flexcompute/tidy3d/blob/develop/tidy3d/components/viz.py#L233)… I don't think we can't fix it in the front-end.

**Comment by tomflexcompute** [comment]
Yeah `plot()` does work correctly in most cases but sometimes there is a warning about mesh being not watertight and the plot would be incorrect, even though the simulation itself is correct. I'll try to see if I can find an example of it.

Regarding the `plot_3d()` issue, I've asked MC to look into it.

**Comment by lucas-flexcompute** [comment]
Maybe @dbochkov-flexcompute or @weiliangjin2021 have better insight here, but I think we cannot guarantee that getting ε from a non-watertight mesh would always succeed. A non-watertight mesh is essentially open, isn't it?

**Comment by caseyflex** [comment]
Yeah I don’t even know if this is a tolerance issue. In the backend we
simply assume the mesh is exactly watertight, understanding that if it is
not, this may lead to permittivity processing errors. Thus it relies on the
frontend usage of trimesh library to check watertightness, only warning if
it may not be watertight. We could play with the frontend tolerance
(currently the trimesh default), but I’m not sure what it should be as
there’s no backend tolerance to compare against.

**Comment by caseyflex** [comment]
I think the relevant tolerance would be tol.merge

https://github.com/mikedh/trimesh/blob/main/trimesh/constants.py

But we need to be a bit careful with it as it identifies vertices. The
default agrees with solidworks.

**Comment by momchil-flex** [comment]
Any follow up on this? What should we do?

**Comment by tomflexcompute** [comment]
MC is looking into it. Their schedule is pretty tight too so not sure if we will have it done within, say, a week or two, but will update here if they make progress.

**Comment by momchil-flex** [comment]
Do we still need to keep this open @tomflexcompute ? I think MC fixed some things.

**Comment by tomflexcompute** [comment]
I think `plot_3d()` still doesn't show stl structures at the moment.

**Comment by majinge7846** [comment]
> I think `plot_3d()` still doesn't show stl structures at the moment.

work on prod now!  you can try it. (need 2.8.3) if this is OK, Could you help to close this?

**Comment by tomflexcompute** [comment]
Verified that it works. Thanks @majinge7846
